modules/core/src/minmax.cpp

```diff
@@ -1510,9 +1510,19 @@ void cv::minMaxIdx(InputArray _src, double* minVal,
 
     Mat src = _src.getMat(), mask = _mask.getMat();
 
+    int _minIdx, _maxIdx;
+    int* min_offset = (cn == 1) ? minIdx : &_minIdx;
+    int* max_offset = (cn == 1) ? maxIdx : &_maxIdx;
     if (src.dims <= 2)
-        CALL_HAL(minMaxIdx, cv_hal_minMaxIdx, src.data, src.step, src.cols, src.rows, src.depth(), minVal, maxVal,
-                 minIdx, maxIdx, mask.data);
+    {
+        CALL_HAL(minMaxIdx, cv_hal_minMaxIdx, src.data, src.step, src.cols*cn, src.rows, src.depth(),
+                    minVal, maxVal, min_offset, max_offset, mask.data);
+    }
+    else if (src.isContinuous())
+    {
+        CALL_HAL(minMaxIdx, cv_hal_minMaxIdx, src.data, 0, (int)src.total()*cn, 1, src.depth(),
+                 minVal, maxVal, min_offset, max_offset, mask.data);
+    }
```

Comment:
In this branch either minIdx and maxIdx need recalculating according to the number of dimensions, or this branch should be disabled if either pointer is non-null.

Reply:
Agreed.